examples/sign-leaf-with-ca.rs

```diff
@@ -0,0 +1,33 @@
+use rcgen::{BasicConstraints, Certificate, CertificateParams, IsCa};
+use std::fs;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+	let mut ca_params: CertificateParams = Default::default();
+	let mut leaf_params: CertificateParams = Default::default();
```

Comment:
Similar to my feedback about the root certificate parameters, let's update the end entity certificate parameters to generate something more closely matching the CABF subscriber certificate profile.

That means:

* Including the authority key identifier ext (e.g. by setting `use_authority_key_identifier_extension` in the params), c.f. baseline reqs ["7.1.2.11.1 Authority Key Identifier"](https://github.com/cabforum/servercert/blob/main/docs/BR.md#712111-authority-key-identifier)
* Adding a SAN extension, c.f. baseline reqs ["7.1.2.7.12 Subscriber Certificate Subject Alternative Name"](https://github.com/cabforum/servercert/blob/main/docs/BR.md#712712-subscriber-certificate-subject-alternative-name)
* Including a key usage extension that asserts only `digitalSignature` (since this is an ECC public key), c.f baseline reqs ["7.1.2.7.11 Subscriber Certificate Key Usage"](https://github.com/cabforum/servercert/blob/main/docs/BR.md#712711-subscriber-certificate-key-usage)
* Including an EKU extensions that asserts `serverAuth`, c.f. baseline reqs ["7.1.2.7.10 Subscriber Certificate Extended Key Usage"](https://github.com/cabforum/servercert/blob/main/docs/BR.md#712710-subscriber-certificate-extended-key-usage)
* Setting the `notBefore` to no more than 48 hours before the date of issuance.
* Setting the `notAfter` to no more than 397 days in the future [c.f. baseline reqs "6.3.2 Certificate operational periods and key pair usage periods"](https://github.com/cabforum/servercert/blob/main/docs/BR.md#632-certificate-operational-periods-and-key-pair-usage-periods)

That would still leave two omissions from the server cert profile:

* The authority information access extension, c.f. baseline reqs ["7.1.2.7.7 Subscriber Certificate Authority Information Access"](https://github.com/cabforum/servercert/blob/main/docs/BR.md#71277-subscriber-certificate-authority-information-access)
* The subscriber certificate policies extension, c.f. baseline reqs ["7.1.2.7.9 Subscriber Certificate Certificate Policies"](https://github.com/cabforum/servercert/blob/main/docs/BR.md#71279-subscriber-certificate-certificate-policies)

We don't have code to emit those two extensions and they're the most specific to the context of a proper root program. I think it's fair to defer those for now :+1: